src/main/java/bio/terra/cli/businessobject/resource/GcsObject.java

```diff
@@ -125,6 +130,27 @@ public String resolve(boolean includeUrlPrefix) {
         : bucketName + "/" + objectName;
   }
 
+  /**
+   * Check if the GCS bucket object is a directory by checking for the presence of a child object
+   * with a trailing slash. There will always be an object present to represent the directory/prefix
+   * itself.
+   */
+  public boolean isPrefix() throws SystemException {
+    try {
+      BucketCow bucketCow =
+          CrlUtils.createStorageCow(Context.requireUser().getPetSACredentials()).get(bucketName);
+      Iterable<BlobCow> objects = bucketCow.list(BlobListOption.prefix(objectName)).getValues();
+      for (BlobCow object : objects) {
+        if (object.getBlobInfo().getName().endsWith("/")) {
```

Comment:
blobInfo has an isDirectory method https://cloud.google.com/java/docs/reference/google-cloud-storage/latest/com.google.cloud.storage.BlobInfo#com_google_cloud_storage_BlobInfo_isDirectory__

Reply:
I had to also pass the `BlobStorageOption.currentDirectory()` option to`list` for it to behave as intended. Thank you!